tests/functional/UseCase/Table/CreateTableFromTimeTravelTest.php

```diff
@@ -0,0 +1,340 @@
+            (new Table())
+                ->setPath($path)
+                ->setTableName($destinationTableName)
+        );
+
+        /** @var int $dateTimeBeforeAlterTimestamp */
+        $dateTimeBeforeAlterTimestamp = strtotime($dateTimeBeforeAlter);
+        $cmd->setTimestamp($dateTimeBeforeAlterTimestamp);
+
+        $handler = new CreateTableFromTimeTravelHandler($this->clientManager);
+
+        /** @var CreateTableFromTimeTravelResponse $response */
+        $response = $handler(
+            $this->projectCredentials,
+            $cmd,
+            []
+        );
+
+        $this->assertSame(4, $response->getTableRowsCount());
+        $this->assertGreaterThan(0, $response->getTableSizeBytes());
+        $this->assertSame(
+            ['col1', 'col2', 'col3'],
+            iterator_to_array($response->getImportedColumns())
+        );
+
+        $sourceRef = new BigqueryTableReflection($bqClient, $bucketDatasetName, $sourceTableName);
+        $this->assertSame(['col1', 'col2', 'col3', 'newColumn'], $sourceRef->getColumnsNames());
+
+        $destinationRef = new BigqueryTableReflection($bqClient, $bucketDatasetName, $destinationTableName);
+        $this->assertSame(['col1', 'col2', 'col3'], $destinationRef->getColumnsNames());
+    }
+
+    public function testInvalidCreateTableFromTimestamp(): void
+    {
+        $sourceTableName = md5($this->getName()) . '_Test_table';
+        $bucketDatasetName = $this->bucketResponse->getCreateBucketObjectName();
+        $bqClient = $this->clientManager->getBigQueryClient($this->projectCredentials);
+        $this->initTestTableAndImportBaseData($bqClient, $bucketDatasetName, $sourceTableName);
+
+        // test create table from non-existing table
+        $cmd = new CreateTableFromTimeTravelCommand();
+        $path = new RepeatedField(GPBType::STRING);
+        $path[] = $bucketDatasetName;
+        $cmd->setSource(
+            (new CreateTableFromTimeTravelCommand\SourceTableMapping())
+                ->setPath($path)
+                ->setTableName('table_should_never_be_created')
+        );
+
+        $destinationTableName = md5($this->getName()) . '_createdFromTimeTravel';
+        $destPath = new RepeatedField(GPBType::STRING);
+        $destPath[] = $bucketDatasetName;
+        $cmd->setDestination(
+            (new Table())
+                ->setPath($destPath)
+                ->setTableName($destinationTableName)
+        );
+
+        /** @var int $timestamp */
+        $timestamp = strtotime(date(DATE_ATOM));
+        $cmd->setTimestamp($timestamp);
+
+        $handler = new CreateTableFromTimeTravelHandler($this->clientManager);
+
+        try {
+            $handler(
+                $this->projectCredentials,
+                $cmd,
+                []
+            );
+            $this->fail('Should fail: Table should never be created.');
+        } catch (NotFoundException $e) {
+            $this->assertSame(404, $e->getCode());
+        }
```

Comment:
To bych odchytil v handleru a vyhodil `ObjectNotFoundException` z common ty exceptions celkově budeme muset poléčit.

Reply:
Upravil som to, jo no chcelo by to asi nejak zjednotit resp. vymysliet nejake obecne exceptions ktore sa budu pak v connection handlovat